sentry_sdk/integrations/spark/spark_driver.py

```diff
@@ -46,18 +55,35 @@ def _sentry_patched_spark_context_init(self, *args, **kwargs):
         _set_app_properties()
 
         with configure_scope() as scope:
-            scope.user = {"id": self.sparkUser()}
-            scope.set_tag("executor.id", self._conf.get("spark.executor.id"))
-            scope.set_tag(
-                "spark.submit.deployMode", self._conf.get("spark.submit.deployMode")
-            )
-            scope.set_tag("driver.port", self._conf.get("spark.driver.port"))
-            scope.set_tag("driver.host", self._conf.get("spark.driver.host"))
-            scope.set_tag("spark_version", self.version)
-            scope.set_tag("app_name", self.appName)
-            scope.set_tag("application_id", self.applicationId)
-
-            scope.set_extra("web_url", self.uiWebUrl)
+
+            @scope.add_event_processor
+            def process_event(event, hint):
+                if Hub.current.get_integration(SparkIntegration) is None:
+                    return event
+
+                event.setdefault("user", {}).setdefault("id", self.sparkUser())
+
+                event.setdefault("tags", {}).setdefault(
+                    "executor.id", self._conf.get("spark.executor.id")
+                )
+                event["tags"].setdefault(
+                    "spark.submit.deployMode", self._conf.get("spark.submit.deployMode")
+                )
+                event["tags"].setdefault(
+                    "driver.host", self._conf.get("spark.driver.host")
+                )
+                event["tags"].setdefault(
```

Comment:
Can you wrap this with `with capture_internal_exceptions()`? This is random access to an API we don't control that might break.